mlrun/api/utils/alembic.py

```diff
@@ -1,50 +1,118 @@
+            self._backup_revision(db_file_path, current_revision)
         alembic.command.upgrade(self._alembic_config, "head")
 
     @staticmethod
-    def _get_db_file_path():
+    def _get_db_file_path() -> str:
         """
         Get the db file path from the dsn.
         Converts the dsn to the file path. e.g.:
         sqlite:////mlrun/db/mlrun.db?check_same_thread=false -> /mlrun/db/mlrun.db
         """
         return mlconf.httpdb.dsn.split("?")[0].split("sqlite:///")[-1]
 
-    def _get_current_revision(self):
-        def print_stdout(text, *arg):
-            self._current_revision = text
+    def _get_current_revision(self) -> typing.Optional[str]:
+
+        # create separate config in order to catch the stdout
+        catch_stdout_config = alembic.config.Config(self._alembic_config_path)
+        catch_stdout_config.print_stdout = self._save_output
+
+        self._flush_output()
+        try:
+            alembic.command.current(catch_stdout_config)
+            return self._alembic_output.strip().replace(" (head)", "")
+        except Exception as exc:
+            if "Can't locate revision identified by" in exc.args[0]:
+
+                # DB has a revision that isn't known to us, extracting it from the exception.
+                return exc.args[0].split("'")[2]
+
+            return None
+
+    def _get_revision_history_list(self) -> typing.List[str]:
+        """
+        Returns a list of the revision history sorted from latest to oldest.
+        """
 
         # create separate config in order to catch the stdout
         catch_stdout_config = alembic.config.Config(self._alembic_config_path)
-        catch_stdout_config.print_stdout = print_stdout
+        catch_stdout_config.print_stdout = self._save_output
+
+        self._flush_output()
+        alembic.command.history(catch_stdout_config)
+        return self._parse_revision_history(self._alembic_output)
+
+    @staticmethod
+    def _parse_revision_history(output: str) -> typing.List[str]:
+        return [line.split(" ")[2].replace(",", "") for line in output.splitlines()]
+
+    @staticmethod
+    def _backup_revision(db_file_path: str, current_version: str):
+        if db_file_path == ":memory:":
+            return
+
+        db_dir_path = pathlib.Path(os.path.dirname(db_file_path))
+        backup_path = db_dir_path / f"{current_version}.db"
+
+        shutil.copy2(db_file_path, backup_path)
+
+    @staticmethod
+    def _downgrade_to_revision(
+        db_file_path: str, current_revision: str, fallback_version: str
+    ):
+        db_dir_path = pathlib.Path(os.path.dirname(db_file_path))
+        backup_path = db_dir_path / f"{fallback_version}.db"
+
+        if not os.path.isfile(backup_path):
+            raise RuntimeError(
+                f"Cannot fall back to revision {fallback_version}, "
+                f"no back up exists. Current revision: {current_revision}"
+            )
+
+        shutil.copy2(backup_path, db_file_path)
```

Comment:
Since this is touching the DB file and a bug here could easily cause a data loss, I try to be very pedant, sorry 😬  - If for some reason we mistakenly did downgrade, the previous DB file might be useful, I suggest adding logic to back it up
```
bla_path = db_dir_path / f"{current_revision}.db"
shutil.copy2(db_file_path, bla_path)
```